modular_skyrat/modules/paycheck_rations/code/quirk.dm

```diff
@@ -25,32 +31,21 @@
 			LOCATION_HANDS = ITEM_SLOT_HANDS,
 		),
 	)
-	account.tracked_ticket_book = WEAKREF(new_ticket_book)
-	account.payday_modifier = 0.5
+	tracked_ticket_book = WEAKREF(new_ticket_book)
+	account.payday_modifier -= 0.5
+	RegisterSignal(account, COMSIG_ON_BANK_ACCOUNT_PAYOUT, PROC_REF(make_ration_ticket))
 	to_chat(client_source.mob, span_notice("You remember to keep close hold of your ticket book, it can't be replaced if lost and all of your ration tickets are placed there!"))
 
-// Edits to bank accounts to make the above possible
-
-/datum/bank_account
-	/// Tracks a linked ration ticket book. If we have one of these, then we'll put tickets in it every payday.
-	var/datum/weakref/tracked_ticket_book
-	/// Tracks if the last ticket we got was for luxury items, if this is true we get a normal food ticket
-	var/last_ticket_luxury = TRUE
-
-/datum/bank_account/payday(amount_of_paychecks, free = FALSE)
-	. = ..()
-	if(!.)
-		return
-	if(isnull(tracked_ticket_book))
+/// Attempts to create a ration ticket book in the card holder's hand, and failing that, the drop location of the card
+/datum/quirk/item_quirk/ration_system/proc/make_ration_ticket(datum/bank_account/account)
+	SIGNAL_HANDLER
+	if(!istype(account))
 		return
-	make_ration_ticket()
 
-/// Attempts to create a ration ticket book in the card holder's hand, and failing that, the drop location of the card
-/datum/bank_account/proc/make_ration_ticket()
 	if(!(SSeconomy.times_fired % 3 == 0))
```

Comment:
This looks like it might be problematic, but that's out of scope

Reply:
It does make it fire every 3rd payout from what I tested (Annoyingly so, had to comment it out for a bit so I can work on testing faster)